sklearn/callback/_callback_context.py

```diff
@@ -207,6 +197,26 @@ def _from_parent(cls, parent_context, *, task_name, task_id, max_subtasks=None):
 
         return new_ctx
 
+    def set_task_info(self, task_name, task_id, max_subtasks=None):
+        """Setter for the attributes relative to the task information.
+
+        Parameters
+        ----------
+        task_name : str
+            The name of the task this context is responsible for.
+
+        task_id : int
+            The id of the task this context is responsible for.
+
+        max_subtasks : int or None, default=None
+            The maximum number of tasks that can be children of the task this context is
+            responsible for. 0 means it's a leaf. None means the maximum number of
+            subtasks is not known in advance.
+        """
+        self.task_name = task_name
+        self.task_id = task_id
+        self.max_subtasks = max_subtasks
+
```

Comment:
we can return it to enable method chaining
```suggestion
        self.task_name = task_name
        self.task_id = task_id
        self.max_subtasks = max_subtasks

        return self
```